introduction-to-doctests-in-haskell/article.md

```diff
@@ -0,0 +1,289 @@
+# Introduction to Doctests in Haskell
+
+If you ever had to skim through the Haskell source code documentation, you might have noticed those lines with fancy `>>>` symbols at the beginning.
+They are called 'doctests' or 'doctest examples'.
+
+To put it simply, doctests are pieces of text within [Haddock comments](https://www.haskell.org/haddock/) that look like interactive Haskell sessions.
+The idea to search for and execute those sessions all over Haskell modules actually comes from the awesome Python's
+[`doctest`](https://en.wikipedia.org/wiki/Doctest) module.
+
+Documenting software is extremely challenging.
+Doctests make the documentation process not only pleasant and effective – it actually allows Haskell functions to speak for themselves – but
+also helps to get the most out of your testing efforts.
+
+## How to define a doctest?
+
+
+Let's look at an example of a doctest. 
```

Comment:
Are there any other doctest libraries? Do they all follow the same syntax?

Reply:
There're a few more libraries. But, AFAIK, they do follow the same syntax